sw/device/lib/dif/dif_rstmgr.h

```diff
@@ -105,6 +105,21 @@ typedef enum dif_rstmgr_reset_info {
   kDifRstmgrResetInfoNdm = (1 << 7),
 } dif_rstmgr_reset_info_t;
 
+/**
+ * All valid enum values for `dif_rstmgr_reset_info_t`.
+ */
+static const dif_rstmgr_reset_info_t kDifRstMgrResetInfoAll[] = {
+    kDifRstmgrResetInfoPor,
+    kDifRstmgrResetInfoLowPowerExit,
+    kDifRstmgrResetInfoSw,
+    kDifRstmgrResetInfoHwReq,
+    kDifRstmgrResetInfoSysRstCtrl,
+    kDifRstmgrResetInfoWatchdog,
+    kDifRstmgrResetInfoPowerUnstable,
+    kDifRstmgrResetInfoEscalation,
+    kDifRstmgrResetInfoNdm,
+};
+
```

Comment:
Can you define this in a c file and declare here as an `extern`?

Reply:
Hmm, I feel like that is kind of asking for the enum and array to get out of sync. I actually am not opposed to just dumping hex values and reading them in the Rust program. I think we can construct the pattern in Rust without much fuss thanks to bindgen.

Reply:
SGTM